src/servers/src/mysql/handler.rs

```diff
@@ -210,59 +274,11 @@ impl<W: AsyncWrite + Send + Sync + Unpin> AsyncMysqlShim<W> for MysqlInstanceShi
         w: StatementMetaWriter<'a, W>,
     ) -> Result<()> {
         let query_ctx = self.session.new_query_context();
-        let (query, param_num) = replace_placeholders(raw_query);
-
-        let statement = validate_query(raw_query).await?;
-
-        // We have to transform the placeholder, because DataFusion only parses placeholders
-        // in the form of "$i", it can't process "?" right now.
-        let statement = transform_placeholders(statement);
-
-        let describe_result = self
-            .do_describe(statement.clone(), query_ctx.clone())
+        let stmt_id = self.prepared_stmts_counter.fetch_add(1, Ordering::Relaxed);
+        let stmt_key = uuid::Uuid::from_u128(stmt_id as u128).to_string();
```

Comment:
The `prepared_stmts_counter` cannot be removed, we still need numeric stmt key because mysql binary protocol for prepare stmt needs it. See https://mariadb.com/kb/en/com_stmt_prepare/, the `COM_STMT_PREPARE_OK`'s stmt id is a 4 bytes int.

Reply:
Got you.